cpp/include/cudf/functions.h

```diff
@@ -944,3 +944,55 @@ gdf_error gdf_order_by(gdf_column** input_columns,
                        size_t       num_inputs,
                        gdf_column*  output_indices,
                        int          flag_nulls_are_smallest);
+
+
+/** 
+ * @brief Scatters the values of a source column into a destination column.
+ * 
+ * Scatters the elements of the source column into the destination column according
+ * to a scatter map such that source[i] will be scattered to destination[ scatter_map[i] ]
+ * 
+ * The number of elements in the scatter_map must equal the number of elements in the source
+ * column. 
+ * 
+ * If any index in scatter_map is outside the range of [0, destination->size), it 
+ * will be ignored.
```

Comment:
Is there a reason to ignore it as opposed to raising an out-of-bounds type error?

Reply:
This is somewhat an artifact of how the internal functionality of gather/scatter is already implemented.

It results from the fact that as a result of an outer join operation, we generate an index of `-1` for a non-match. Then when we go to gather from the source columns into the destination, we ignore the `-1` indices. 

I suppose we don't need to expose this detail externally in case it ever changes.

Reply:
Note that the alternative isn't that we will detect an out-of-bounds error and gracefully return you an error code, you'll just get a segfault from the underlying Thrust algorithms we're using.

Reply:
Changing it to "undefined" as you have is more appropriate. I like.